src/libs/actions/Report.js

```diff
@@ -1543,6 +1464,112 @@ function renameReport(reportID, reportName) {
         .finally(() => Onyx.set(ONYXKEYS.IS_LOADING_RENAME_POLICY_ROOM, false));
 }
 
+/**
+ * @param {Number} reportID
+ * @param {Object} action
+ */
+function viewNewReportAction(reportID, action) {
+    const newMaxSequenceNumber = action.sequenceNumber;
+    const isFromCurrentUser = action.actorAccountID === currentUserAccountID;
+
+    // When handling an action from the current users we can assume that their
+    // last read actionID has been updated in the server but not necessarily reflected
+    // locally so we must first update it and then calculate the unread (which should be 0)
+    if (isFromCurrentUser) {
+        setLocalLastRead(reportID, newMaxSequenceNumber);
+    }
+
+    const updatedReportObject = {
+        // Use updated lastReadSequenceNumber, value may have been modified by setLocalLastRead
+        // Here deletedComments count does not include the new action being added. We can safely assume that newly received action is not deleted.
+        unreadActionCount: newMaxSequenceNumber - (lastReadSequenceNumbers[reportID] || 0) - ReportActions.getDeletedCommentsCount(reportID, lastReadSequenceNumbers[reportID] || 0),
+    };
+
+    Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`, updatedReportObject);
```

Comment:
Thinking from an offline perspective. Let's say I'm offline and eventually I receive the new report action. If the 10-second check returns early we won't run this code and thus won't update the `unreadActionCount`, right?

Reply:
If you are offline then you can't get any Pusher updates at all. There is no "eventually" - you missed it and it's never coming.

Reply:
If you're asking what happens when we come back online - then we would fetch the reports which should update the `unreadActionCount` for any reports. If you manage to miss a report action update because you're offline and then come back online before the 10 second window - I guess you'd see a notification - which seems alright.

Reply:
Ah okay, thanks for clarifying!